build/Makefile

```diff
@@ -384,6 +384,7 @@ STATIC_PROVIDERLIB_SRCFILES += \
 	$(PROVIDER_DIR)/support/mycnf-paths.cpp \
 	$(PROVIDER_DIR)/support/productdependencies.cpp \
 	$(PROVIDER_DIR)/support/sqlauth.cpp \
+	$(PROVIDER_DIR)/support/mycnf-scanner.cpp \
```

Comment:
What we are introducing now seemed some sort of mysql specific "support" functionality - like mycnf-paths.cpp and sqlauth.cpp. so I moved the files here.

Moreover, for some reason that I do not quite understand, [sqlbinding.cpp](https://github.com/Microsoft/MySQL-Provider/blob/master/source/code/providers/support/sqlbinding.cpp#L136) in this folder is reading the cnf files. That line gets executed  at least by the myauth executable (which also runs with root permissions, btw). At some point in future, sqlbinding.cpp may need a subset of the scanner functionality that we are introducing, because mysql_options does not support "!include" and "!includedir"

If you think my above concerns are invalid, I will move this to [omi_prexec](https://github.com/Microsoft/MySQL-Provider/tree/master/source/code/omi_preexec) folder